### PR TITLE
Update to spec base 09

### PR DESCRIPTION
**Pull request by gregsdennis**
This should add some tests for https://www.ietf.org/archive/id/draft-ietf-jsonpath-base-09.html as well as rename existing tests so that the language is more aligned with the spec.

Relates to https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/357 and https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/358.

This is not at all meant to be comprehensive.  ~Also, I have a few bugs in my implementation around detection of nulls, that I'm still working out, but everything else passes.~

**Edit**: I have fixed the bugs in my implementation.  All tests now pass.

### CODE REVIEW

**Comment by f3ath** [comment]
I wonder if it would be more manageable to have the renames in a separate PR.

**Comment by gregsdennis** [comment]
I figured having a separate commit would suffice.

File: cts.json [hunk 1]
```diff
@@ -1720,12 +1720,174 @@
...
+      "name": "filter, equals null, absent from data",
+      "selector" : "$[?@.a==null]",
+      "document" : [{"d": "e"}, {"a":"c", "d": "f"}],
+      "result": [
+      ]
+    },
+    {
+      "name": "filter, equals true",
+      "selector" : "$[?@.a==true]",
+      "document" : [{"a": true, "d": "e"}, {"a":"c", "d": "f"},],
```
Comment (glyn) [inline cts.json]: Extraneous comma.

File: cts.json [hunk 2]
```diff
@@ -2066,14 +2066,55 @@
...
+        {"a":"d", "d": "f"}
+      ]
+    },
+    {
+      "name": "filter, not expression",
+      "selector" : "$[?!(@.a=='b')]",
+      "document" : [{"a": "a", "d": "e"}, {"a":"b", "d": "f"}, {"a":"d", "d": "f"}],
+      "result": [
+        {"a": "a", "d": "e"},
+        {"a":"d", "d": "f"}
```
Comment (glyn) [inline cts.json:2100]: Missing space.

**Review by glyn — APPROVED** [review]
LGTM, thanks. One formatting nit.

File: cts.json (hunk 1 above)
Comment (gregsdennis) [reply cts.json]: I thought I got this.... 🤔

**Pull request merged**